Src/Newtonsoft.Json/Linq/JsonPath/QueryExpression.cs

```diff
@@ -257,5 +271,10 @@ private bool EqualsWithStringCoercion(JValue value, JValue queryValue)
 
             return string.Equals(currentValueString, queryValueString, StringComparison.Ordinal);
         }
+        private bool EqualsWithoutStringCoercion(JValue value, JValue queryValue)
+        {
+            // Looking at the impl I believe it considers type and value and may be euqivalent to javascript's identity op. I think. :/
+            return value.Equals(queryValue);
```

Comment:
This will need to handle:
1 === 1.0: True

Reply:
The test suite needs work, re my previous comment. Asking around for docs on coercion rules, will update.